jcl/src/java.base/share/classes/com/ibm/jit/JITHelpers.java

```diff
@@ -566,6 +566,22 @@ public char getCharFromArrayByIndex(Object obj, int index) {
 		}
 	}
 
+	public boolean canEncodeAsLatin1(byte[] array, int start, int length) {
+		int count = 0;
+		int index = start << 1;
```

Comment:
I presume the expectation is that JIT will either recognize this method and substitute a native implementation or optimize this reasonably well: if the latter, might that not be better served by having one loop variable instead of two? Perhaps like this:
```java
		int index = start << 1;
		if (!IS_BIG_ENDIAN) {
			index += 1;
		}
		for (int end = index + (length << 1); index < end; index += 2) {
			if (array[index] != 0) {
				return false;
			}
		}
		return true;
```

Reply:
I was concerned about overflow but I believe this is fine. I've made the update.